src/track/trackrecord.cpp

```diff
@@ -129,25 +130,50 @@ bool TrackRecord::updateSourceSynchronizedAt(
     }
     setSourceSynchronizedAt(sourceSynchronizedAt);
     m_headerParsed = sourceSynchronizedAt.isValid();
-    DEBUG_ASSERT(isSourceSynchronized());
     return true;
 }
 
-bool TrackRecord::isSourceSynchronized() const {
-    // This method cannot be used to update m_headerParsed
-    // after modifying m_sourceSynchronizedAt during a short
-    // moment of inconsistency. Otherwise the debug assertion
-    // triggers!
+TrackRecord::SourceSyncStatus TrackRecord::checkSourceSyncStatus(
+        const FileInfo& fileInfo) const {
+    // This method cannot be used to update m_headerParsed after modifying
+    // m_sourceSynchronizedAt during a short moment of inconsistency.
+    // Otherwise the debug assertion triggers!
     DEBUG_ASSERT(m_headerParsed ||
             !getSourceSynchronizedAt().isValid());
+    // Legacy fallback: The property sourceSynchronizedAt has been added later.
+    // Files that have been added before that time and that have never been
+    // re-imported will only have that legacy flag set while sourceSynchronizedAt
+    // is still invalid.
+    if (!m_headerParsed) {
+        // Enforce initial import of metadata if it hasn't succeeded
+        // at least once yet.
+        return SourceSyncStatus::Void;
+    }
     if (getSourceSynchronizedAt().isValid()) {
-        return true;
+        const QDateTime fileSourceSynchronizedAt =
+                MetadataSource::getFileSynchronizedAt(fileInfo.asQFileInfo());
+        if (fileSourceSynchronizedAt.isValid()) {
+            if (getSourceSynchronizedAt() < fileSourceSynchronizedAt) {
+                return SourceSyncStatus::Outdated;
+            } else {
+                if (getSourceSynchronizedAt() > fileSourceSynchronizedAt) {
+                    kLogger.warning()
```

Comment:
Why is this a warning?
We should provide a info that explains that.

Reply:
https://github.com/mixxxdj/mixxx/pull/4218/commits/4576f9bd32baeb192c4ef11318b5b90179d59d23